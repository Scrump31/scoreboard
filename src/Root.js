import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';

import store from './reducers/index';

export default ({ children }) => (
  <Provider store={createStore(store)}>{children}</Provider>
);
