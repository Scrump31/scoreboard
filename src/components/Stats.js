import React from 'react';

import { Table } from 'react-bootstrap';

export default function Stats(props) {
  return (
    <Table className="stats">
      <tbody>
        <tr className="total-players-td-row">
          <td><strong>Players:</strong></td>
          <td>{props.totalPlayers}</td>
        </tr>
        <tr>
          <td><strong>Total Points:</strong></td>
          <td>{props.totalPoints}</td>
        </tr>
      </tbody>
    </Table>
  );
}
Stats.propTypes = {
  totalPlayers: React.PropTypes.number,
  totalPoints: React.PropTypes.number,
};
Stats.defaultProps = {
  totalPlayers: 0,
  totalPoints: 0,
};
