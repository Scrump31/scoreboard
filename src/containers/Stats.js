import React from 'react';
import { connect } from 'react-redux';
import _ from 'lodash';
import { Table } from 'react-bootstrap';

const Stats = ({ totalPlayers, totalPoints }) => (
  <Table className="stats">
    <tbody>
      <tr className="total-players-td-row">
        <td>
          <strong>Players:</strong>
        </td>
        <td>{totalPlayers}</td>
      </tr>
      <tr>
        <td>
          <strong>Total Points:</strong>
        </td>
        <td>{totalPoints}</td>
      </tr>
    </tbody>
  </Table>
);

Stats.propTypes = {
  totalPlayers: React.PropTypes.number,
  totalPoints: React.PropTypes.number,
};
Stats.defaultProps = {
  totalPlayers: 0,
  totalPoints: 0,
};

const mapStateToProps = ({ players }) => ({ players });

export default connect(mapStateToProps)(Stats);
