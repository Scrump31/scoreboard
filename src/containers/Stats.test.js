import React from 'react';
import { mount } from 'enzyme';
import Root from '../Root';
import Stats from './Stats';

describe('<Stats />', () => {
  it('renders without crashing', () => {
    const component = mount(
      <Root>
        <Stats />
      </Root>,
    );
    expect(component).toMatchSnapshot();
    component.unmount();
  });
});
