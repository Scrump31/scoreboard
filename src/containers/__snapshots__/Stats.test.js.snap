// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stats /> renders without crashing 1`] = `
<Component>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(Stats)>
      <Stats
        dispatch={[Function]}
        players={
          Array [
            Object {
              "id": "1",
              "name": "Player One",
              "score": 31,
            },
            Object {
              "id": "2",
              "name": "Player Two",
              "score": 35,
            },
            Object {
              "id": "3",
              "name": "Player Three",
              "score": 42,
            },
          ]
        }
      >
        <Table
          bordered={false}
          bsClass="table"
          className="stats"
          condensed={false}
          hover={false}
          responsive={false}
          striped={false}
        >
          <table
            className="stats table"
          >
            <tbody>
              <tr
                className="total-players-td-row"
              >
                <td>
                  <strong>
                    Players:
                  </strong>
                </td>
                <td>
                  3
                </td>
              </tr>
              <tr>
                <td>
                  <strong>
                    Total Points:
                  </strong>
                </td>
                <td>
                  108
                </td>
              </tr>
            </tbody>
          </table>
        </Table>
      </Stats>
    </Connect(Stats)>
  </Provider>
</Component>
`;
